internal/api/auth.go

```diff
@@ -93,23 +93,23 @@ func (a *API) maybeLoadUserOrSession(ctx context.Context) (context.Context, erro
 	claims := getClaims(ctx)
 
 	if claims == nil {
-		return ctx, unauthorizedError("invalid token: missing claims")
+		return ctx, forbiddenError(ErrorCodeBadJWT, "invalid token: missing claims")
 	}
 
 	if claims.Subject == "" {
-		return nil, unauthorizedError("invalid claim: missing sub claim")
+		return nil, forbiddenError(ErrorCodeBadJWT, "invalid claim: missing sub claim")
 	}
 
 	var user *models.User
 	if claims.Subject != "" {
 		userId, err := uuid.FromString(claims.Subject)
 		if err != nil {
-			return ctx, badRequestError("invalid claim: sub claim must be a UUID").WithInternalError(err)
+			return ctx, badRequestError(ErrorCodeBadJWT, "invalid claim: sub claim must be a UUID").WithInternalError(err)
```

Comment:
qq: Why do we use `badRequestError` here but `forbiddenError` for line 123?

I thought about it loosely using the questions:
 > Should the user/client be someone else to get a different status code?

We'd get a different status code if the user was what is stated in the sub claim

Then I considered the case for 401:
>  Are the credentials the user/client sending missing or invalid in form, structure, encoding?

Yes, the sub claim is invalid.


Then I thought about it for session:
 > Should the user/client be someone else to get a different status code?

We'd get a different status code (200) if the session_id exists

Then I considered the case for 401:
>  Are the credentials the user/client sending missing or invalid in form, structure, encoding?

I realize that it's part art as well so just wondering if there's further thought behind it. OK to go with current set up - just wondering what the thinking process was like here in order to use for ref when deciding on the label for future cases

Reply:
Because there's something the developer can do to fix it -- use a proper JWT with the correct claims. If the JWT was not signed correctly, or had insufficient permissions then it's 403. But it's 400 if the data in the JWT can't be recognized.